cirq/sim/mux.py

```diff
@@ -183,3 +200,74 @@ def sample_sweep(program: Union[circuits.Circuit, schedules.Schedule],
                               seed=prng)
         trial_results.append(measurements)
     return trial_results
+
+
+def final_density_matrix(
+        program: Union[circuits.Circuit, ops.Gate, ops.OP_TREE, schedules.
+                       Schedule],
+        *,
+        noise: 'cirq.NOISE_MODEL_LIKE' = None,
+        initial_state: Union[int, Sequence[Union[int, float, complex]], np.
```

Comment:
I have a PR defining `cirq.STATE_VECTOR_LIKE` ( https://github.com/quantumlib/Cirq/pull/2376 ). We should define an analogous `cirq.DENSITY_MATRIX_LIKE` for this case.

Reply:
Sure, that makes sense. By the way, there seems to be no state_vector() method defined for DensityMatrixTrialResult - is that by design?

Reply:
A density matrix is more general than a state vector, so it's not possible to define such a method. (Or rather, it would need to have different semantics such as sampling a state vector from the density matrix, or failing if the density matrix did not correspond to a pure state.)

Reply:
Thanks, that makes sense. Common suffix TrialResult makes it look like all the result classes should have a common interface (or a common ancestor), but they don't, which can be a bit confusing.

Reply:
This might be better to do as a separate PR.

Reply:
That works.